extension/sourcesdk/NextBot/NextBotInterface.cpp

```diff
@@ -44,4 +157,186 @@ void INextBot::DebugConColorMsg( NextBotDebugType debugType, const Color &color,
+			 object->GetHealth())
+		{
+			return true;
+		}
+
+		if (FClassnameIs(const_cast<CBaseEntityHack*>( object ), "func_breakable_surf"))
+		{
+			return true;
+		}
+
+		// TO-DO: Fix
+		/*if (dynamic_cast<const CBreakableProp*>(object) != nullptr)
+		{
+			return true;
+		}*/
+	}
+
+	return false;
+}
+
+bool INextBot::IsRangeLessThan(CBaseEntityHack* subject, float range) const
+{
+	Vector botPos;
+	CBaseEntityHack* bot = const_cast<INextBot *>(this)->GetEntity();
+	if (!bot || !subject)
+		return 0.0f;
+	bot->CollisionProp()->CalcNearestPoint(subject->WorldSpaceCenter(), &botPos);
+	float computedRange = subject->CollisionProp()->CalcDistanceFromPoint(botPos);
+	return computedRange < range;
+}
+
+bool INextBot::IsRangeLessThan(const Vector &pos, float range) const
+{
+	Vector to = pos - GetPosition();
+	return to.IsLengthLessThan(range);
+}
+
+bool INextBot::IsRangeGreaterThan(CBaseEntityHack *subject, float range) const
+{
+	Vector botPos;
+	CBaseEntityHack *bot = const_cast<INextBot *>(this)->GetEntity();
+	if (!bot || !subject)
+		return true;
+
+	bot->CollisionProp()->CalcNearestPoint( subject->WorldSpaceCenter(), &botPos);
+	float computedRange = subject->CollisionProp()->CalcDistanceFromPoint(botPos);
+	return computedRange > range;
+}
+
+bool INextBot::IsRangeGreaterThan(const Vector &pos, float range) const
+{
+	Vector to = pos - GetPosition();
+	return to.IsLengthGreaterThan(range);
+}
+
+float INextBot::GetRangeTo(CBaseEntityHack *subject) const
+{
+	Vector botPos;
+	CBaseEntityHack* bot = const_cast<INextBot *>(this)->GetEntity();
+	if (!bot || !subject)
+		return 0.0f;
+
+	bot->CollisionProp()->CalcNearestPoint(subject->WorldSpaceCenter(), &botPos);
+	float computedRange = subject->CollisionProp()->CalcDistanceFromPoint(botPos);
+	return computedRange;
+}
+
+float INextBot::GetRangeTo(const Vector &pos) const
+{
+	Vector to = pos - GetPosition();
+	return to.Length();
+}
+
+float INextBot::GetRangeSquaredTo(CBaseEntityHack *subject) const
+{
+	Vector botPos;
+	CBaseEntityHack* bot = const_cast<INextBot*>(this)->GetEntity();
+	if (!bot || !subject)
+		return 0.0f;
+
+	bot->CollisionProp()->CalcNearestPoint(subject->WorldSpaceCenter(), &botPos);
+	float computedRange = subject->CollisionProp()->CalcDistanceFromPoint(botPos);
+	return computedRange * computedRange;
+}
+
+float INextBot::GetRangeSquaredTo(const Vector &pos) const
+{
+	Vector to = pos - GetPosition();
+	return to.LengthSqr();	
+}
+
+bool INextBot::IsDebugging(unsigned int type) const
+{
+	// TO-DO: Fix
+	/*if (TheNextBots().IsDebugging(type))
+	{
+		return TheNextBots().IsDebugFilterMatch(this);
+	}*/
+	return false;
+}
+
+bool INextBot::IsDebugFilterMatch(const char *name) const
+{
+	if (!Q_strnicmp(name, GetDebugIdentifier(), Q_strlen(name)))
+	{
+		return true;
+	}
+
+	// TO-DO: fix
+	/*CTeam *team = GetEntity()->GetTeam();
+	if (team && !Q_strnicmp(name, team->GetName(), Q_strlen(name)))
+	{
+		return true;
+	}
+	return false;*/
```

Comment:
Need to return something here.

```suggestion
	*/
	return false;
```